src/uct/cuda/base/cuda_md.c

```diff
@@ -167,34 +169,38 @@ uct_cuda_base_query_attributes(uct_cuda_copy_md_t *md, const void *address,
         return UCS_ERR_INVALID_ADDR;
     }
 
-    if (is_managed) {
-        mem_info->type = UCS_MEMORY_TYPE_CUDA_MANAGED;
-    } else {
-        mem_info->type = UCS_MEMORY_TYPE_CUDA;
-
-        /* Synchronize for DMA */
-        cu_err = cuPointerSetAttribute(&value, CU_POINTER_ATTRIBUTE_SYNC_MEMOPS,
-                                       (CUdeviceptr)address);
-        if (cu_err != CUDA_SUCCESS) {
-            cuGetErrorString(cu_err, &cu_err_str);
-            ucs_warn("cuPointerSetAttribute(%p) error: %s", address,
-                    cu_err_str);
-        }
-    }
-
     status = uct_cuda_base_get_sys_dev(cuda_device, &mem_info->sys_dev);
     if (status != UCS_OK) {
         return status;
     }
 
+    if (is_managed) {
+        mem_info->type         = UCS_MEMORY_TYPE_CUDA_MANAGED;
+        mem_info->base_address = base_address;
+        mem_info->alloc_length = alloc_length;
```

Comment:
can we set address/length here instead of base_address/alloc_length, and then avoid the initialization in line 150-151?
also pls add comment why we don't want to check alloc size for managed memory

Reply:
can we use address/length here?